File: solutions/6/sol.go
package main

import (
	"fmt"
)

func sumOfSquares(n int) int {
	sum := 0
	for num := 1; num <= n; num++ {
		sum += num * num
	}
	return sum
}

func squareOfSum(n int) int {
	sum := n * (n + 1) / 2
	return sum * sum
}

func main() {
	fmt.Println(sumOfSquares(10))
	fmt.Println(squareOfSum(10))
}

File: solutions/6/sol.ts
const sumOfSquares: (n: number) => number = (n) => {
  let sum = 0;
  for (let num = 1; num <= n; num++) {
    sum += num * num;
  }
  return sum;
}

const squareOfSum: (n: number) => number = (n) => {
  const sum = n * (n + 1) / 2;
  return sum * sum;
}

const getAnswer: (n: number) => number = (n) => {
  return squareOfSum(n) - sumOfSquares(n)
}

console.log(getAnswer(10))